### PR TITLE
Share code for receive and transmit_and_receive.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/resources/rmt_esp32.cc [hunk 1]
```diff
@@ -362,6 +333,44 @@ PRIMITIVE(transmit_and_receive) {
...
+  int receive_items_length = receive_bytes.length() / 4;
+
+  return transmit_and_receive(tx_channel, rx_channel,
+                              transmit_items, transmit_items_length,
+                              receive_items, receive_items_length,
+                              max_output_len,
+                              receive_timeout_ms,
+                              process);
+}
+
```
Comment (kasperl) [inline src/resources/rmt_esp32.cc:373]: Drop extra newline.

File: src/resources/rmt_esp32.cc [hunk 2]
```diff
@@ -362,6 +333,44 @@ PRIMITIVE(transmit_and_receive) {
...
+  ARGS(RMTResource, tx, RMTResource, rx, Blob, transmit_bytes, Blob, receive_bytes, int, max_output_len, int, receive_timeout_ms)
+  if (transmit_bytes.length() % 4 != 0) INVALID_ARGUMENT;
+  if (receive_bytes.length() % 4 != 0) INVALID_ARGUMENT;
+
+  rmt_channel_t rx_channel = rx->channel();
+  rmt_channel_t tx_channel = tx->channel();
+
+  const rmt_item32_t* transmit_items = reinterpret_cast<const rmt_item32_t*>(transmit_bytes.address());
+  const rmt_item32_t* receive_items = reinterpret_cast<const rmt_item32_t*>(receive_bytes.address());
+  int transmit_items_length = transmit_bytes.length() / 4;
```
Comment (kasperl) [inline src/resources/rmt_esp32.cc:363]: Should you pass null items if the lengths are zero?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/resources/rmt_esp32.cc (hunk 2 above)
Comment (floitsch) [reply src/resources/rmt_esp32.cc:363]: Good find.
Switched to checking for the length in the shared function instead, this way I avoid having a check here.

File: src/resources/rmt_esp32.cc (hunk 1 above)
Comment (floitsch) [reply src/resources/rmt_esp32.cc:373]: done.

**Pull request merged**